### PR TITLE
Edits to Readme and Dynamic Multithreading

### LINKED ISSUE #25: [Bug] FFmpeg is not listed in dependencies for Ubuntu
## Description

`FFmpeg` is needed to use MediaGrapher with videos and as needed by dependency `ffmpeg-python==0.2.0`. This system dependency should be listed in your `README.md`.

## To Reproduce

1. Run `python mediagrapher.py <video link>`

## Expected behavior

The program should be able to process all frames of the video accordingly.

## Screenshots

Command line error message:
```bash
(.venv) raynelfss@DESKTOP-CONEHPG:~/home/JedProject/MediaGrapher$ python mediagrapher.py https://youtu.be/d6qCbdXqsOs
[youtube] Extracting URL: https://youtu.be/d6qCbdXqsOs
[youtube] d6qCbdXqsOs: Downloading webpage
[youtube] d6qCbdXqsOs: Downloading ios player API JSON
[youtube] d6qCbdXqsOs: Downloading android player API JSON
[youtube] d6qCbdXqsOs: Downloading m3u8 information
[info] d6qCbdXqsOs: Downloading 1 format(s): 22
[download] Destination: input/input.mp4
[download] 100% of   12.56MiB in 00:00:01 at 6.42MiB/s
Processing video...
Getting video metadata...
Traceback (most recent call last):
  File "/home/raynelfss/home/JedProject/MediaGrapher/mediagrapher.py", line 264, in <module>
    main()
  File "/home/raynelfss/home/JedProject/MediaGrapher/mediagrapher.py", line 258, in main
    process_video(media, os.path.join("input", "frames"), OUTPUT)
  File "/home/raynelfss/home/JedProject/MediaGrapher/mediagrapher.py", line 202, in process_video
    metadata = get_video_metadata(video_path)
  File "/home/raynelfss/home/JedProject/MediaGrapher/mediagrapher.py", line 173, in get_video_metadata
    probe = ffmpeg.probe(video_path)
  File "/home/raynelfss/home/JedProject/MediaGrapher/.venv/lib/python3.10/site-packages/ffmpeg/_probe.py", line 20, in probe
    p = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
  File "/usr/lib/python3.10/subprocess.py", line 971, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.10/subprocess.py", line 1863, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'ffprobe'
```

**Desktop (please complete the following information):**

- OS: Ubuntu 22.04 (WSL)
- Version: n/A

### PR DESCRIPTION
**Pull request by hMRZQ21**
# Pull Request

## Describe your changes
- Edited Readme to include:
1. Installing requirements.txt
2. ffmpeg dependency
3. Command line documentation

- Added new optional argument to command line `-p threads`. A print statement will confirm the number of threads being utilized.

It defaults to `MAX_THREADS` and minimum is 1 thread implemented using `range()`:

```
MAX_THREADS = os.cpu_count()
parser.add_argument('-p', '--threads', type=int, choices=range(1, MAX_THREADS+1), default=MAX_THREADS,
                    help="Number of threads utilized on the CPU. (default: MAX_THREADS)")
```
Testing that it works- 

My machine runs on 8 cores, 16 threads and by inputting 0 we see the CLI returns a detailed error message:
image

## Issue ticket number and link

Resolves #25 

## Topic

- [x] Enhancement / New Feature
- [x] Bug Fix
- [x] Documentation
- [ ] Test
- [ ] Breaking Change

## Supporting Docs

Attach any supporting documents for your pull request such as images, code snippets, etc.

Time difference on the following [video](https://www.youtube.com/watch?v=P75q_3Qlrqg):

4 threads: 14 mins 17 secs
16 threads: 6 mins 20 secs

### CODE REVIEW

File: mediagrapher.py
Comment (jedrm) [inline mediagrapher.py:1]: Please test if given a number that is larger than the number of available threads and lower than 1 thread

**Review by jedrm — COMMENTED** [review]
Check the comment under mediagrapher.py

File: mediagrapher.py
Comment (jedrm) [reply mediagrapher.py:1]: You can probably put a set of choices for the thread count using `range`

Comment (hMRZQ21) [reply mediagrapher.py:1]: My machine is capable of 16 threads at most. I tried inputting 24 threads and this is the output:
image

Trying 0 threads:
image

Trying a negative number of threads:
image

The one liner I have seems to correctly force the thread count in the desired range. I can also try using `range` in a new PR.

**Review by jedrm — APPROVED** [review]
LGTM

**Pull request merged**